### DISCUSSION TITLE
Installer with user defined installation path

### DISCUSSION THREAD

**Discussion by Quirulum** [pulibrary/ExcelAlmaLookup#22] [Ideas]
Is it possible to get an installer that allows to define the installation path of the plugin during the installation process? Currently, the installation path is the user's predefined AppData folder, which means that the plugin can only be used by that specific user. We would like to make it available to all users (e.g. by installing it to "C:\Program Files" or "C:\Program Files (x86)").

**Comment by tventimi** [comment]
Hello,

That's a great suggestion, and I think it shouldn't be a difficult feature to add.  I will look into it and let you know the progress on this thread.

Tom Ventimiglia
Princeton University Library

**Reply by tventimi** [reply]
Hello, just wanted to follow up on this request, as I have looked into it more.  While it is possible to allow a custom installation path, the installation process also involves adding registry entries so that Excel knows to automatically load the add-in at startup.  For a global installation, each user's registry profile would need to be modified.  Is this something you would want as well, or just to place the macro itself in a common location so that individual users can opt in to loading it?

**Reply by Quirulum** [reply]
For starters it would be grate to place the macro and dependencies, like the .dll files, in a common location so other users can opt in. I don't know if the macro still works without changing the registry entries. I already tried to copy paste the folder with the installed files to a new location and manually edit the registry entries after the installation, but that did not work. I don't know if I missed some registry entries... but the macro still looked for the .dll files at the old AppData path.

**Reply by tventimi** [reply]
Thanks for your patience as I have been investigating this.  I am happy to let you know that I have released a beta version of the plugin (v1.3.6) with an installer that allows installation to a custom location.  Could you please try it and let me know if it works for you?  Here is the link to the beta installer:

https://github.com/pulibrary/ExcelAlmaLookup/releases/download/v1.3.6-beta/CatalogLookupInstaller.exe

Please note that "Look Up In Catalog" button in the Excel ribbon will only be activated for the user running the installer.  After installation, other users can activate the button by performing the following steps

1) Open Excel and go to File > Options > Add-Ins
2) At the bottom of the window, next to "Manage:", make sure "Excel Add-Ins" is selected, and click "Go..."
3) Click "Browse..." and select the file "CatalogLookup.xlam" that was previously installed to the custom directory.

So really, the purpose of this installer is to make the plugin files and DLLs available from a common location, not to fully activate the plugin for all users at once.  Also, individual users will still have their own custom settings (URL lists, Field Sets).

Please let me know if this would meet your needs.  I know there are still some steps required of individual users, but hopefully it will save some work and allay any security concerns with non-admins running the installer EXE.  Once you have confirmed this works for you, and once I have included these features in an official release, I will update the documentation accordingly. Thanks again for the suggestion!

Tom

**Reply by Quirulum** [reply]
Hi Tom

Thanks a lot! The installer works as expected and the Add-In can be installed to a custom directory.
Unfortunately I just discovered, that the Add-In does not work with the M365 Excel. The Add-In gets installed and the "Look Up In Catalog" button is active, but running it does not work, nothing happens. There is no error or any other message. Running the Add-In in Excel 2021 works perfectly fine. I can open an other issue for that, because I think its not directly related to our discussion here.

Romano

**Reply by tventimi** [reply]
Hi Romano,

I'm sorry to hear it isn't working on M365, but I am surprised because this is the version of Excel I am using.  Did this issue occur with previous versions of the plugin before v1.3.6-beta?  Does it always occur, or only under certain circumstances (e.g. installing in a custom directory)?  Also, can you provide the specific version number (found under File > Account > About Excel)?  Thanks, and hopefully I can get to the bottom of this.

Tom

**Reply by Quirulum** [reply]
Hi Tom,
It is the same problem with the previous version of the plugin to. The Add-In gets installed and the "Look Up In Catalog" button is active, but running it does not work, nothing happens. We are using the following Version of Excel "Microsoft 365 MSO (Version 2501 Build 16.0.18429.20044) 64 Bit". 
If the Add-In works with your M365 Version of Excel, it might be a change in our company policy or settings related to Excel Add-Ins I don't know of. I have to check that further.

**Reply by tventimi** [reply]
I am running version 2412 (and don't seem to have the ability to upgrade to 2501), but I'm not sure if a minor version change would be the issue anyway.  I'm not sure how long you have been using the plugin, but was there any previous version where the issue did not occur?  If so, what happens if you try to reinstall that version? (A full list of installers from past versions can be found here):

https://github.com/pulibrary/ExcelAlmaLookup/releases

**Reply by tventimi** [reply]
I have released v1.3.6 with the ability to specify a custom installation directory.  Feel free to open a new ticket under "Issues" regarding the other bug you are experiencing.  And thanks again for the great suggestion!
